### PR TITLE
CDAP-8778 Check for admin privilege in hbase for cdap if transaction …

### PR DESCRIPTION
**Pull request by rohitsinha54**
…pruning is enabled

Issue: https://issues.cask.co/browse/CDAP-8778
Build:

### CODE REVIEW

File: cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java [hunk 1]
```diff
@@ -83,7 +83,7 @@
     ROW, ROWCOL, NONE
   }
 
-  private static final Logger LOG = LoggerFactory.getLogger(HBaseTableUtil.class);
+  protected static final Logger LOG = LoggerFactory.getLogger(HBaseTableUtil.class);
```
Comment (anew) [inline cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java]: why protected?

File: cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java [hunk 2]
```diff
@@ -66,5 +70,41 @@ public void run() {
...
+          if (cConf.getBoolean(Constants.Startup.TX_PRUNE_ACL_CHECK, false)) {
+            LOG.info("Found {} to be set to true. Continuing with cdap master startup even though global admin check " +
+                       "returned false", Constants.Startup.TX_PRUNE_ACL_CHECK);
+            return;
+          }
+          StringBuilder builder = new StringBuilder("Transaction pruning is enabled and cdap does not have global " +
+                                                      "admin privileges in HBase. Global admin privileges for cdap " +
+                                                      "is required for transaction pruning. " +
+                                                      "Either disable transaction pruning or grant global admin " +
+                                                      "privilege to cdap in HBase.");
```
Comment (anew) [inline cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java]: Or at your own risk override this check by setting TX_PRUNE_ACL_CHECK in cdap-site.xml.

File: cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java [hunk 3]
```diff
@@ -66,5 +70,41 @@ public void run() {
...
+            builder.append(" CDAP will not be able determine if it has global admin privilege in HBase.");
+            builder.append(" After granting global admin privilege please set ");
+            builder.append(Constants.Startup.TX_PRUNE_ACL_CHECK);
+          }
+          throw new RuntimeException(builder.toString());
+        } catch (IOException e) {
+          throw new RuntimeException("Unable to determines cdap privileges as global admin in HBase.");
+        }
+      }
+      LOG.info("Hbase authorized is disabled. Skipping global admin check.");
```
Comment (anew) [inline cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java]: authorization.

Comment (anew) [reply cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java]: or pruning is disabled?

**Comment by anew** [comment]
is this tested on a cluster?

**Comment by rohitsinha54** [comment]
Yes. Its tested on cluster.

File: cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java (hunk 1 above)
Comment (rohitsinha54) [inline cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java]: we are logging the exception if caught one from the compat modules.

File: cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java [hunk 4]
```diff
@@ -90,7 +90,10 @@ public void run() {
                                                       "admin privileges in HBase. Global admin privileges for cdap " +
                                                       "is required for transaction pruning. " +
```
Comment (anew) [inline cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java]: is/are

File: cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java [hunk 5]
```diff
@@ -90,7 +90,10 @@ public void run() {
...
                                                       "Either disable transaction pruning or grant global admin " +
-                                                      "privilege to cdap in HBase.");
+                                                      "privilege to cdap in HBase or at your own risk override this " +
```
Comment (anew) [inline cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java]: perhaps "at your own risk" is a little harsh without explaining the risk. Let's just say, You can override this check by setting....

File: cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java [hunk 6]
```diff
@@ -104,7 +107,7 @@ public void run() {
           throw new RuntimeException("Unable to determines cdap privileges as global admin in HBase.");
         }
       }
-      LOG.info("Hbase authorized is disabled. Skipping global admin check.");
     }
+    LOG.info("Hbase authorization is disabled. Skipping global admin check.");
```
Comment (anew) [inline cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java]: ok, so the reason we need the global admin check is that Pruning is on, and HBase authorization is on, too. To make this understandable to the user, that's what you want to log, above when you determine that you want to check, right now you only say pruning is on. And down here you only say HBase auth is on. Both of these are incomplete and should instead contain both pieces of information.

**Comment by poornachandra** [comment]
LGTM

**Pull request merged**